optaplanner-benchmark/src/main/java/org/optaplanner/benchmark/impl/ranking/SingleBenchmarkRankingComparator.java

```diff
@@ -31,8 +31,8 @@
     public int compare(SingleBenchmarkResult a, SingleBenchmarkResult b) {
         return new CompareToBuilder()
                 .append(b.isFailure(), a.isFailure()) // Reverse, less is better (redundant: failed benchmarks don't get ranked at all)
-                .append(b.getUninitializedVariableCount(), a.getUninitializedVariableCount()) // Reverse, less is better
-                .append(a.getScore(), b.getScore(), resilientScoreComparator)
+                .append(b.getAverageUninitializedVariableCount(), a.getAverageUninitializedVariableCount()) // Reverse, less is better
+                .append(a.getAverageScore(), b.getAverageScore(), resilientScoreComparator)
```

Comment:
That was also in mind, but I think we need to make this change anyway.
That way the API is clear that it's an average.

I think we can count the number of users that have a custom ranking can be counted on 1 hand.
But even if that wasn't the case: I think impl changes (so non public api changes!) that impact user code should not influence our decision on whether or not to make them significantly. Long-term it's our API that matters and at some point that impl api will become public api so it should be as perfect as possible by then. But we need to clearly document every potential user impacting change in the Upgrade recipe, to at least minimize the pain inflicted on our users.